shared/util.go

```diff
@@ -1008,6 +1008,15 @@ func EscapePathFstab(path string) string {
 	return r.Replace(path)
 }
 
+func SetProgressMetadata(metadata map[string]string, stage, displayPrefix string, percent, speed int64) {
+	// stage, percent, speed sent for API callers.
+	metadata["progress_stage"] = stage
+	metadata["progress_percent"] = strconv.FormatInt(percent, 10)
+	metadata["progress_speed"] = strconv.FormatInt(speed, 10)
+	// <stage>_progress with formatted text sent for lxc cli.
+	metadata[stage+"_progress"] = fmt.Sprintf("%s: %d%% (%s/s)", displayPrefix, percent, GetByteSizeString(speed, 2))
```

Comment:
It seems like the <stage> will be something that clients will rely on so we should make very sure to make it generic enough that we don't break it when we change those functions.
Is `create_image_from_container_compress` not a bit long and overly specific? If at all it feels like this should be:
```
"progress_"+stage
```
so callers can consistently check for the `progress_` prefix.

Reply:
The use of the '_progress' suffix here was suggested by stgraber@ in https://github.com/lxc/lxd/pull/5444#issuecomment-459307607 to make this code work:
https://github.com/lxc/lxd/blob/115158b5069fca5e5e9f9be9d1fb48712c217013/lxc/utils/progress.go#L188

As for the 3 names that I've chosen for the stages:
 - create_container_from_image_unpack
 - create_image_from_container_tar
 - create_image_from_container_compress

I'm happy to change them, and in particular to have them shorter like:
 - unpack
 - tar
 - compress

I just thought that there may be more than 1 place that we do something like 'tar' or 'compress', so it could work out better to be specific about each different case.

Reply:
@joelhockey if you think that'd work for you, can you send a follow-up branch which does that?

Reply:
I just send https://github.com/lxc/lxd/pull/5467 which I think does what you are suggesting.